lib/oeqa/selftest/cases/updater_native_non_ostree.py

```diff
@@ -0,0 +1,13 @@
+from updater_native import SotaToolsTests, GeneralTests
+
+
+class SotaToolsTestsNonOSTree(SotaToolsTests):
+    pass
+
+
+class GeneralTestsNonOSTree(GeneralTests):
+    def test_feature_sota(self):
+        pass
+
+    def test_feature_usrmerge(self):
+        pass
```

Comment:
Are you intending to fill these in? I'm not quite sure what would be meaningful here.

Reply:
No, These are tests that we want to avoid in the non-ostere, since we inherit from GeneralTests class these two cases are not relevant, so we override them with an empty body. This looks ugly but it's how we can reuse existing test cases for the non-ostree.

Reply:
> it's how we can reuse existing test cases for the non-ostree.

As we discussed yesterday, I don't think reusing test cases makes sense. The tests should be independent so that we can run them regardless of the current environment setup.

Reply:
That's actually what we do,  with the current approach we can run the tests for both cases independently, we just reuse some common code and test cases from existing classes just to avoid duplication code (test cases) which is general for both cases `ostrere` and `non-ostree`.
So we can now separately run, for example, `updater_qemux86_64` and `updater_qemux86_64_non_ostree`, but not together for the same setup.
I'm just not sure that I understand you correctly, what do you mean when you say, - The tests should be independent..., they are independent!

Reply:
> what do you mean when you say, - The tests should be independent..., they are independent!

Yes, I see that this term is confusing! I mean they should be independent of your current build environment. You (ideally) should haven't to preconfigure anything to run the tests.

> we can now separately run, for example, `updater_qemux86_64` and `updater_qemux86_64_non_ostree`, but not together for the same setup.

That indicates that the tests are *not* independent of the current environment setup.

> we just reuse some common code and test cases from existing classes just to avoid duplication code (test cases) which is general for both cases `ostrere` and `non-ostree`.

We should not need separate test cases for the OSTree and non-OSTree cases. We don't need to re-test everything for the non-OSTree case. All we need to add is a single new test that reconfigures the environment to build and boot and non-OSTree image and then reset everything back to how it was before. The amount of reconfiguration is a bit more than what we've done for other tests, but each of the tests already does some amount of this already.

Reply:
Ok, I think I catch an idea, previously I've considered performing those tests for the `non-ostree` env setup, so I'm going to rework, using one env and extend existing test files with the cases for `non-ostree`.
One more question, what do you mean, we do not need to re-test everything, actually, we haven't some special cases for `non-ostree`, anyway, we have to retest some of them, not all, cases for 
`non-sotree`, like a test of provisioning, etc.

Reply:
> what do you mean, we do not need to re-test everything, actually, we haven't some special cases for `non-ostree`, anyway, we have to retest some of them, not all, cases for `non-sotree`, like a test of provisioning, etc.

I don't think it's necessary to retest all of the variations. I think just covering the basic (shared credential) provisioning case is fine. We only need a single test to prove that we can still build and boot an image without OSTree.